dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3763,240 +3838,300 @@ Sedp::Reader::data_received(const DCPS::ReceivedDataSample& sample)
+Sedp::LivelinessReader::data_received_i(const DCPS::ReceivedDataSample& sample,
+  const DCPS::EntityId_t& entity_id,
+  DCPS::Serializer& ser,
+  const DCPS::Extensibility extensibility)
+{
+  ACE_UNUSED_ARG(extensibility);
 
-      DCPS::unique_ptr<DiscoveredPublication_SecurityWrapper> wdata_secure(new DiscoveredPublication_SecurityWrapper);
+  const DCPS::MessageId id = static_cast<DCPS::MessageId>(sample.header_.message_id_);
+  const bool full_message = !sample.header_.key_fields_only_;
 
-      if (!ParameterListConverter::from_param_list(data, *wdata_secure, wdata_secure->type_info)) {
-        ACE_ERROR((LM_ERROR,
-                   ACE_TEXT("(%P|%t) ERROR: Sedp::Reader::data_received - ")
-                   ACE_TEXT("failed to convert from ParameterList ")
-                   ACE_TEXT("to DiscoveredPublication_SecurityWrapper\n")));
-        return;
-      }
+  if (entity_id == ENTITYID_P2P_BUILTIN_PARTICIPANT_MESSAGE_WRITER && full_message) {
+    DCPS::unique_ptr<ParticipantMessageData> data(new ParticipantMessageData);
+    if (!(ser >> *data)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("ERROR: Sedp::Reader::data_received - ")
+        ACE_TEXT("failed to deserialize data\n")));
+      return;
+    }
+    sedp_.task_.enqueue(id, move(data));
 
-      wdata_secure->have_ice_agent_info = false;
-      ICE::AgentInfoMap ai_map;
-      if (!ParameterListConverter::from_param_list(data, ai_map)) {
-        ACE_ERROR((LM_ERROR,
-                   ACE_TEXT("(%P|%t) ERROR: Sedp::Reader::data_received - ")
-                   ACE_TEXT("failed to convert from ParameterList ")
-                   ACE_TEXT("to ICE Agent info\n")));
-        return;
-      }
-      ICE::AgentInfoMap::const_iterator pos = ai_map.find("DATA");
-      if (pos != ai_map.end()) {
-        wdata_secure->have_ice_agent_info = true;
-        wdata_secure->ice_agent_info = pos->second;
-      }
-      sedp_.task_.enqueue(id, move(wdata_secure));
+#ifdef OPENDDS_SECURITY
+  } else if (entity_id == ENTITYID_P2P_BUILTIN_PARTICIPANT_MESSAGE_SECURE_WRITER && full_message) {
+
+    DCPS::unique_ptr<ParticipantMessageData> data(new ParticipantMessageData);
+
+    if (!(ser >> *data)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("ERROR: Sedp::Reader::data_received - ")
+        ACE_TEXT("failed to deserialize data\n")));
+      return;
+    }
+    sedp_.task_.enqueue_participant_message_secure(id, move(data));
 #endif
+  }
+}
 
-    } else if (entity_id == ENTITYID_SEDP_BUILTIN_SUBSCRIPTIONS_WRITER) {
-      ParameterList data;
-      if (!decode_parameter_list(sample, ser, extensibility, data)) {
-        ACE_ERROR((LM_ERROR, ACE_TEXT("ERROR: Sedp::Reader::data_received - ")
-                   ACE_TEXT("failed to deserialize data\n")));
-        return;
-      }
+void
+Sedp::SecurityReader::data_received_i(const DCPS::ReceivedDataSample& sample,
+  const DCPS::EntityId_t& entity_id,
+  DCPS::Serializer& ser,
+  const DCPS::Extensibility extensibility)
+{
+ACE_UNUSED_ARG(sample);
```

Comment:
```suggestion
  ACE_UNUSED_ARG(sample);
```
These could be moved to the preprocessor "else" block